### ISSUE TITLE
False positive RV_RETURN_VALUE_IGNORED with @CanIgnoreReturnValue

### ISSUE THREAD

**Issue by lasselindqvist** [spotbugs/spotbugs#1046]
https://github.com/spotbugs/spotbugs/pull/538 added support for reading `com.google.errorprone.annotations.CanIgnoreReturnValue` annotations and not producing unnecessary warnings when those are used, but my experience was the opposite.

guava 18.0 has the following method
```
  /**
   * Stores a key-value pair in the multimap.
   *
   * @param key key to store in the multimap
   * @param value value to store in the multimap
   * @return {@code true} always
   */
  @Override public boolean put(@Nullable K key, @Nullable V value) {
    return super.put(key, value);
  }
```

guava 20.0 has the following method

```
  /**
   * Stores a key-value pair in the multimap.
   *
   * @param key key to store in the multimap
   * @param value value to store in the multimap
   * @return {@code true} always
   */
  @CanIgnoreReturnValue
  @Override
  public boolean put(@Nullable K key, @Nullable V value) {
    return super.put(key, value);
  }
```

When updating from 18.0 Spotbugs suddenly started producing unnecessary warnings when calling this method from our code. The same happened with other similar guava methods that had the annotation added to them.

We used spotbugs-maven-plugin 3.1.3 which depends on spotbugs 3.1.2
The pull request was merged and released in 3.1.2 (https://github.com/spotbugs/spotbugs/blob/master/CHANGELOG.md#312---2018-02-24)

Is it possible that the check logic is backwards in the implementation and warnings are produced when the annotation exists?
